core/app/models/spree/tax/item_adjuster.rb

```diff
@@ -0,0 +1,32 @@
+module Spree
+  module Tax
+    class ItemAdjuster
+      include TaxHelpers
+
+      def initialize(item, options = {})
+        @item = item
+        @order = @item.order
+        # set the instance variable so `TaxRate.match` is only called when necessary
+        @order_rates = options[:order_tax_rates]
+      end
+
+      def adjust!
+        return unless order_tax_zone
+        # Using .destroy_all to make sure callbacks fire
+        @item.adjustments.tax.destroy_all
+
+        TaxRate.store_pre_tax_amount(@item, rates_for_item)
+
+        rates_for_item.each do |rate|
+          rate.adjust(order_tax_zone, @item)
+        end
+      end
+
+      private
+
+      def rates_for_item
+        @rates ||= applicable_rates.select { |rate| rate.tax_category == @item.tax_category }
```

Comment:
I find it good practice to name caching instance variables after the methods name and use an underscore to make it more visible, that this is not a normal instance variable. So, the name should be `@_rates_for_item` IMO. 

And we should also use the `item` `attr_reader` here.